hdwallet/src/lib.rs

```diff
@@ -79,8 +82,27 @@ impl HDLattice {
         if path.is_empty() {
             return Ok(self.master_key);
         }
-        let entries = path.split('/');
+
+        // Check if this is a wormhole path
+        let (is_wormhole, remaining_path) = if path.starts_with("w/") {
+            (true, &path[2..])
+        } else {
+            (false, path)
+        };
```

Comment:
Feels like we should either chop off the first character for both cases or for none?

Reply:
let's do that in a separate pr tho